### PR TITLE
Final 1.x updates

### PR DESCRIPTION
**Pull request by jeffmay**
@vodopis @chrissalij-r @nicksovich @htmldoug @jtsmith0107 

This is the last update before I make a major version shift to 2.0.0 with support for Play Json 2.4.6.

Updates to play-json-ops:
- Add builder syntax for AbstractJsonOps.extractTypeKey for better type inference
- Add support for object serialization with AbstractJsonOps.extractTypeKey.pure
- Handle ClassCastExceptions for mismatched types
- Add new AbstractJsonOps.extractTypeKey.usingKeyObject
- Added tests
- Updated Scala version

### CODE REVIEW

File: playJsonOps/src/main/scala/play/api/libs/json/ops/AbstractJsonOps.scala [hunk 1]
```diff
@@ -300,25 +359,29 @@ abstract class TypeKeyExtractor[T] private[ops] {
...
+    *
+    * @return the json object to be combined with the output json
    */
   def writeKeyToJson(key: Key): JsObject = keyFormat writes key
 
   /**
    * Throws an [[UnrecognizedTypeKey]] exception with all the appropriate arguments.
-   * @param key the parsed key that has no corresponding [[OFormat]] for the specific
+    *
+    * @param key the parsed key that has no corresponding [[OFormat]] for the specific
```
Comment (jeffmay) [inline playJsonOps/src/main/scala/play/api/libs/json/ops/AbstractJsonOps.scala:461]: Might as well fix the indentation everywhere.

**Comment by jeffmay** [comment]
@patrick-bohan and @slustbader might want to check this one out as well.

**Comment by jeffmay** [comment]
Also, @nicksovich

File: build.sbt [hunk 2]
```diff
@@ -72,6 +76,3 @@ lazy val playJsonTests = project in file("playJsonTests") settings(common: _*) s
     "me.jeffmay" %% "scalacheck-ops" % "1.1.0"
```
Comment (jeffmay) [inline build.sbt]: We can upgrade to scalacheck-ops 1.2.0

**Comment by ghost** [comment]
LGTM

**Comment by vodopis** [comment]
LGTM

**Pull request merged**